### PR TITLE
fix(api): load InventoryValues in emissions by scope

### PR DESCRIPTION
**Pull request by lemilonkh**
### What change is being made?

Modify the emissions calculation to include third-party emissions by loading `InventoryValues` scoped to each inventory in the results service and update handling in the API route layer.

### Why are these changes being made?

This change is necessary to ensure accurate emissions reporting by including emissions data from third-party sources when internal activity values are absent. By refactoring the API route and results service, the system can now calculate emissions comprehensively, addressing potential data gaps and improving reporting accuracy.

> Is this description stale? Ask me to generate a new description by commenting `/korbit-generate-pr-description`

### CODE REVIEW

File: app/src/backend/ResultsService.ts [hunk 1]
```diff
@@ -426,6 +433,38 @@ function convertEmissionsToStrings(
...
+    const co2eq = value.co2eq ?? 0n;
+    const totalEmissions = bigIntToDecimal(co2eq);
+    return {
+      activityTitle: "third-party", // TODO use data source name or SubCategory name?
+      scopes: {
+        [scopeName]: bigIntToDecimal(co2eq),
+      },
+      totalEmissions,
+      percentage: 100,
+    };
```
[the remark this answers is not in the document]
Comment (lemilonkh) [reply app/src/backend/ResultsService.ts:463]: @korbit-ai there is only one activity for the whole inventory value in the case of third party data associated with a whole GPC reference number.

File: app/src/backend/ResultsService.ts [hunk 2]
```diff
@@ -426,6 +436,36 @@ function convertEmissionsToStrings(
...
```
Comment (isaaccodekill) [inline app/src/backend/ResultsService.ts:463]: Shouldn't this be limited to inventory values with a data source id? Especially if the goal is to get a scope breakdown of thirdparty emissions

**Review by isaaccodekill — APPROVED** [review]
Amazing work

File: app/src/backend/ResultsService.ts [hunk 3]
```diff
@@ -426,6 +436,36 @@ function convertEmissionsToStrings(
...
+    if (!scopeName) {
+      throw new createHttpError.InternalServerError(
+        "Scope name not found for inventory value " + value.id,
+      );
+    }
+
+    const co2eq = value.co2eq ?? 0n;
+    const totalEmissions = bigIntToDecimal(co2eq);
+    return {
+      activityTitle: "third-party", // TODO use data source name or SubCategory name?
```
Comment (thehighestprimenumber) [inline app/src/backend/ResultsService.ts:457]: Dino wanted to call this "Mixed activities"

**Review by thehighestprimenumber — APPROVED** [review]
just a comment about naming

File: app/src/backend/ResultsService.ts (hunk 2 above)
Comment (lemilonkh) [reply app/src/backend/ResultsService.ts]: Good point, done :+1: I'm also using the dataset name instead of `third-party` now.

**Pull request merged**